api/v1/cluster_types.go

```diff
@@ -1919,6 +1920,14 @@ type MonitoringConfiguration struct {
 	// +kubebuilder:default:=false
 	// +optional
 	EnablePodMonitor bool `json:"enablePodMonitor,omitempty"`
+
+	// The list of metric relabelings for the `PodMonitor`. Applied to samples before ingestion.
+	// +optional
+	PodMonitorMetricRelabelConfigs []*monitoringv1.RelabelConfig `json:"podMonitorMetricRelabelings,omitempty"`
+
+	// The list of relabelings for the `PodMonitor`. Applied to samples before scraping.
+	// +optional
+	PodMonitorRelabelConfigs []*monitoringv1.RelabelConfig `json:"podMonitorRelabelings,omitempty"`
```

Comment:
we could even think to add the whole podmonitorspec here probably, overriding only what we care about (the port and the label selector mainly)

Reply:
Sounds good!
Asking as a Go newbie, is there a facility to ease with merging, taking into account that PodMetricsEndpoint are defined in a list and users may want to target one element to reach RelabelConfigs? Thanks

Reply:
No, there is not unfortunately, but given that we are opinionated on the endpoint to hit, we could just enforce our own for that. But this implementation is fine, we can move to the full spec later on if we want to